api/openai/request.go

```diff
@@ -136,6 +136,16 @@ func updateRequestConfig(config *config.Config, input *schema.OpenAIRequest) {
 		}
 	}
 
+	if len(input.Tools) > 0 {
+		for _, tool := range input.Tools {
+			input.Functions = append(input.Functions, tool.Function)
+		}
+	}
+
+	if input.ToolsChoice != nil {
+		input.FunctionCall = input.ToolsChoice
```

Comment:
It's not the same, there is the same "one level up" abstraction:

`function_call` is  `{"name": "my_function"}`

`tool_choice` is `{"type": "function", "function": {"name": "my_function"}}`

Also, the JSON response is also different. I guess you get the "new" response when you provided the tools array instead of functions:

```json
"choices": [
    {
      "index": 0,
      "message": {
        "role": "assistant",
        "content": null,
        "tool_calls": [
          {
            "id": "call_abc123",
            "type": "function",
            "function": {
              "name": "get_current_weather",
              "arguments": "{\n\"location\": \"Boston, MA\"\n}"
            }
          }
        ]
      },
      "logprobs": null,
      "finish_reason": "tool_calls"
    }
  ]
```

Reply:
This should be fixed now, thanks for pointing it out!